src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedObj: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const value = String(item[key]);
+
+    if (!(value in groupedObj)) {
+      groupedObj[value] = [];
+    }
```

Comment:
I heard that better use method hasOwnProperty.

Reply:
Pro hint: [Use OR assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR_assignment) instead of `if`.